### ISSUE TITLE
Branching of two graphs breaks pipeline

### ISSUE THREAD

**Issue by a-hanf** [mlr-org/mlr3pipelines#519]
Hello,
I encountered this issue when building a branching pipeline. I am not sure whether I am using the pipeline correctly. Is this intended behaviour?

I demonstrate this here by first building two graphs, which can be trained and behave as expected:
```
library(mlr3)
library(mlr3pipelines)
library(mlr3learners)

graph1 = pipeline_robustify(tsk("iris"), lrn("classif.svm"), impute_missings = TRUE)
graph1$set_names(graph1$ids(), paste("graph1", graph1$ids(), sep = "."))
graph1 = graph1 %>>% lrn("classif.svm")

graph2 = pipeline_robustify(tsk("iris"), lrn("classif.ranger"))
graph2$set_names(graph2$ids(), paste("graph2", graph1$ids(), sep = "."))
graph2 = graph2 %>>% lrn("classif.ranger")

graph1$train(tsk("iris"))
graph2$train(tsk("iris"))
# no errors here
```
When I put them in a single pipeline using `ppl("branch")`, I get an error during training:
```
pipeline = ppl("branch", graphs = list(graph1, graph2))
pipeline$train(tsk("iris"))
# breaks
```
```
Error in check_item(data[[idx]], typetable[[operation]][[idx]], varname = sprintf("%s %s (\"%s\") of PipeOp %s",  : 
  Assertion on 'input 1 ("input") of PipeOp graph1.imputehist
Conversion from given data to Task produced message:
no applicable method for 'as_task' applied to an object of class "list".' failed: Must inherit from class 'Task', but has class 'list'.
```
The error can be fixed by adding a `po("nop")` at the beginning of the first graph:
```
graph1 = po("nop") %>>% graph1
pipeline = ppl("branch", graphs = list(graph1, graph2))
pipeline$train(tsk("iris"))
# works again
```
This is a simple workaround, but I am not sure it should be used this way.
Thank you!

**Comment by mb706** [comment]
Thanks for the report, I will investigate this.

Btw, you can use `graph1$update_ids(prefix = "graph1.")` to prefix things. the operation is in-place, so you can do
```r
graph1 = pipeline_robustify(tsk("iris"), lrn("classif.svm"), impute_missings = TRUE)$update_ids(prefix = "graph1.")
```
in one line

**Comment by mb706** [comment]
Was actually fixed by #673
